### ISSUE TITLE
largeString shouldMatchFile "workflow.yaml"

### ISSUE THREAD

**Issue by jmfayard** [kotest/kotest#2945]
**Please describe the feature you'd like to see including any solutions in mind if you have any**

To compare very large strings, it's easier to write the expected content in a separate file.

https://github.com/krzema12/github-actions-kotlin-dsl/blob/6fc0b3f279ee7372e084545245a08d9e51f36317/library/src/test/kotlin/it/krzeminski/githubactions/IntegrationTest.kt#L25-L39

```kotlin
class IntegrationTest : FunSpec({
    setupShouldMatchFile(
        baseDirectory = File("src/test/resources/integration"),
        normalize = { it.replace("\r\n", "\n") }, // test shouldn't fail on CI running on windows
    )

    test("workflowTest : toYaml()") {
        workflowTest.toYaml() shouldMatchFile "workflowTest.yaml"
    }

}
```

When the test fails, the error message is:

```
Files differ in /Users/jmfayard/IdeaProjects/github-actions-kotlin-dsl/library/src/test/resources/integration
Expected: expected/workflowTest.yaml
Actual:     actual/workflowTest.yaml 
``` 

That tells me exactly what I need to use IntelliJ Diff tool, the perfect tool to check what happens and fix it.

<img width="1018" alt="github-actions-kotlin-dsl_–_workflowTest_yaml__et_New_Issue_·_kotest_kotest" src="https://user-images.githubusercontent.com/459464/166074129-14596fc5-934f-40f8-93ff-5b10ecc629ea.png">

The alternative to put the very long string inside the test is ugly

https://github.com/krzema12/github-actions-kotlin-dsl/blob/main/library/src/test/kotlin/it/krzeminski/githubactions/IntegrationTest.kt#L77-L150

Especially modifying the large expected string is hard.

My implementation is done at https://github.com/krzema12/github-actions-kotlin-dsl/blob/6fc0b3f279ee7372e084545245a08d9e51f36317/library/src/test/kotlin/it/krzeminski/githubactions/kotest/ShouldMatchFile.kt

I fall back to use `this shouldBe file.readText()` on the CI (for me GitHub Actions) because there I can't easily compare the content of the actual and the expected file.

**Comment by sksamuel** [comment]
Hi. I'm not sure what you're asking for :)
Are you suggesting we add a "string" shouldBe file matcher?

**Comment by jmfayard** [comment]
Yes that was the suggestion :)

**Comment by nor-ek** [comment]
Can assign me

**Comment by Kantis** [comment]
I typically define a method for reading from the test resources like:
```kotlin
fun readResource(path: String): String =
   object {}.javaClass.getResource(path)?.readText() ?: error("Failed to load resource at $path")
```

then your test could simply be:

```kotlin
actual shouldBe readResource("/integration/expectedWorkflow.yaml")
```

If the strings differ by content you will get a link for diffing:

<img width="1524" alt="image" src="https://user-images.githubusercontent.com/316857/166568399-59bd8966-d196-4d7b-a970-972c1f8662bf.png">

Which opens the diff, without having to select files manually.

In which scenario would the suggestion add extra convenience?

**Comment by jmfayard** [comment]
@Kantis `readResource` is nice ; when I use IntelliJ diffing tool, I do not only _see_ the diff, I can also directly _edit_ the expected file, accepting or rejecting chunks of the diff. For large files that's quite convenient.
<img width="1585" alt="github-actions-kotlin-dsl_–_DockerImage_kt" src="https://user-images.githubusercontent.com/459464/166634892-0c492fa7-f6d5-4194-aea3-13a302d5bf58.png">

**Comment by Kantis** [comment]
Ah.. gotcha! That's pretty neat. Could we somehow maybe have `str shouldMatchResource("/integration/workflow.yaml")`  w/o any setup (always assume from test resources) and if that particular assertion fails, write `str` to a temp file and somehow make it easy to diff? Would be awesome with a link from the console. Perhaps doable with the IntelliJ plugin somehow?

**Comment by jmfayard** [comment]
Hello yes that would be awesome if the intellij plugin could enable that

**Comment by bzareba** [comment]
Hi, I've submitted a PR for this but I was wondering how to make the `Expected` file path clickable in IntelliJ. The only way that I found is to get absolute path of that resource file, but it points to the `build` directory which make no sense in scope of editing this in diff window. 

Also `Actual` file is not clickable on dry run (when that file doesn't exist in build directory). 

![image](https://user-images.githubusercontent.com/15802693/195894618-94c10c70-c4a7-475c-8451-05b2245c22f3.png)
